src/main/java/games/battlelore/BattleloreForwardModel.java

```diff
@@ -0,0 +1,245 @@
+package games.battlelore;
+
+import core.AbstractForwardModel;
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.components.GridBoard;
+
+import core.interfaces.IGamePhase;
+import core.rules.Node;
+import games.battlelore.actions.AttackUnitsAction;
+import games.battlelore.actions.MoveUnitsAction;
+import games.battlelore.actions.PlayCommandCardAction;
+import games.battlelore.actions.SkipTurnAction;
+import games.battlelore.cards.CommandCard;
+import games.battlelore.components.MapTile;
+import games.battlelore.components.Unit;
+
+import utilities.Utils;
+
+import java.util.*;
+
+
+public class BattleloreForwardModel extends AbstractForwardModel {
+    @Override
+    protected void _setup(AbstractGameState initialState) {
+        Random random = new Random(initialState.getGameParameters().getRandomSeed());
+        BattleloreGameParameters gameParams = (BattleloreGameParameters) initialState.getGameParameters();
+        BattleloreGameState gameState = (BattleloreGameState)initialState;
+        BattleloreData _data = gameState.getData();
+
+        for (int i = 0; i  < gameState.playerCount; i++) {
+            if (gameState.playerCount != 2) {
+                System.out.println("3 and more players are not supported");
+            }
+        }
+
+        //Init player hands
+        int hexHeight = gameParams.hexHeight;
+        int hexWidth = gameParams.hexWidth;
+
+        //Game Area Initialization
+        gameState.gameBoard = new GridBoard<MapTile>(hexWidth, hexHeight);
+        gameState.unitTypes = new ArrayList<>();
+        gameState.unitTypes = _data.getUnits();
+
+        int tileId = 0;
+        for (int x = 0; x < gameState.gameBoard.getWidth(); x++) {
+            for(int y = 0; y < gameState.gameBoard.getHeight(); y++) {
+                gameState.gameBoard.setElement(x, y, new MapTile(x, y, new ArrayList<Unit>(), null, tileId));
+                tileId++;
+            }
+        }
+
+        PutLearningScenarioUnits(gameState);
+        gameState.setGamePhase(BattleloreGameState.BattleloreGamePhase.CommandAndOrderStep);
+    }
+
+    @Override
+    protected void _next(AbstractGameState currentState, AbstractAction action) {
+        BattleloreGameState state = (BattleloreGameState) currentState;
+        BattleloreGameParameters gameParams = (BattleloreGameParameters) currentState.getGameParameters();
+        action.execute(currentState);
+
+        int playerId = state.getCurrentPlayer();
+        Unit.Faction playerFaction = playerId == Unit.Faction.Dakhan_Lords.ordinal() ?
+                Unit.Faction.Dakhan_Lords : Unit.Faction.Uthuk_Yllan;
+
+        switch (state.getGamePhase().toString()) {
+            case "CommandAndOrderStep":
```

Comment:
This should use the enum, not a string, much easier to mess up the string name and I believe it might be a costlier comparison too.

Reply:
done in part 1